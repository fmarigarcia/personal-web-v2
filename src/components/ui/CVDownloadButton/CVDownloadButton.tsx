import { useTranslation } from 'react-i18next';
import { HiArrowDownTray } from 'react-icons/hi2';
import { Button } from '@components/ui';

interface CVDownloadButtonProps {
    className?: string;
    variant?: 'primary' | 'secondary';
}

export const CVDownloadButton: React.FC<CVDownloadButtonProps> = ({
    className = '',
    variant = 'secondary',
}) => {
    const { t } = useTranslation();

    const handleDownload = () => {
        // Create a link element and trigger download
        const link = document.createElement('a');
        link.href = '/CV Francisco Marí.pdf';
        link.download = 'CV Francisco Marí.pdf';
        link.target = '_blank';
        link.rel = 'noopener noreferrer';
        document.body.appendChild(link);
        link.click();
        document.body.removeChild(link);
    };

    return (
        <Button
            variant={variant}
            onClick={handleDownload}
            className={className}
            aria-label={t('experience.downloadCV')}
        >
            <span className="flex items-center gap-2">
                <HiArrowDownTray
                    className="w-5 h-5"
                    data-testid="download-icon"
                />
                {t('experience.downloadCV')}
            </span>
        </Button>
    );
};
