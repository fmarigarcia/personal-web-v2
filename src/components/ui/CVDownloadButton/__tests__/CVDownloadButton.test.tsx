import { render, screen } from '@testing-library/react';
import { CVDownloadButton } from '../CVDownloadButton';

// Mock react-i18next
jest.mock('react-i18next', () => ({
    useTranslation: () => ({
        t: jest.fn((key: string) => {
            if (key === 'experience.downloadCV') return 'Download CV';
            return key;
        }),
    }),
}));

// Mock the Button component
jest.mock('@components/ui', () => ({
    Button: ({
        children,
        onClick,
        className,
        ...props
    }: Record<string, unknown>) => (
        <button
            onClick={onClick as () => void}
            className={className as string}
            {...props}
        >
            {children as React.ReactNode}
        </button>
    ),
}));

// Mock the icon
jest.mock('react-icons/hi2', () => ({
    HiArrowDownTray: () => <span data-testid="download-icon">â†“</span>,
}));

describe('CVDownloadButton', () => {
    it('should render correctly', () => {
        render(<CVDownloadButton />);

        expect(screen.getByRole('button')).toBeInTheDocument();
        expect(screen.getByText('Download CV')).toBeInTheDocument();
        expect(screen.getByTestId('download-icon')).toBeInTheDocument();
    });

    it('should apply custom className', () => {
        render(<CVDownloadButton className="test-class" />);

        const button = screen.getByRole('button');
        expect(button).toHaveClass('test-class');
    });

    it('should have correct aria-label', () => {
        render(<CVDownloadButton />);

        const button = screen.getByRole('button');
        expect(button).toHaveAttribute('aria-label', 'Download CV');
    });

    it('should match snapshot', () => {
        const { container } = render(<CVDownloadButton />);
        expect(container.firstChild).toMatchSnapshot();
    });
});
