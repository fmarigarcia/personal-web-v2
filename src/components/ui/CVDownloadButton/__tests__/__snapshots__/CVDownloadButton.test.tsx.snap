// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CVDownloadButton should match snapshot 1`] = `
<button
  aria-label="Download CV"
  class=""
  variant="secondary"
>
  <span
    class="flex items-center gap-2"
  >
    <span
      data-testid="download-icon"
    >
      â†“
    </span>
    Download CV
  </span>
</button>
`;
